### PR TITLE
tsh cleanups

### PR DESCRIPTION
**Pull request by zmb3**
This PR includes some minor cleanups to `tsh`:

- allow for generating CPU/memory profiles via hidden flags
- remove old `tsh config-proxy` command
- minor refactor to make `tsh play` code easier to follow

### CODE REVIEW

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -420,6 +423,8 @@ func Run(args []string, opts ...cliOption) error {
 	app.Flag("proxy", "SSH proxy address").Envar(proxyEnvVar).StringVar(&cf.Proxy)
 	app.Flag("nocache", "do not cache cluster discovery locally").Hidden().BoolVar(&cf.NoCache)
 	app.Flag("user", fmt.Sprintf("SSH proxy user [%s]", localUser)).Envar(userEnvVar).StringVar(&cf.Username)
+	app.Flag("memprofile", "Write memory profile to file").StringVar(&memProfile).Hidden()
+	app.Flag("cpuprofile", "Write CPU profile to file").StringVar(&cpuProfile).Hidden()
```
Comment (greedy52) [inline tool/tsh/tsh.go:427]: 👍 very handy, thanks for adding them.

nit: personal preference `cpu-profile` vs `cpuprofile`. I feel likely I would type `--cpu-profile` unconsciously as other flags have dashes dividing the words.

**Review by greedy52 — APPROVED** [review]

File: tool/tsh/tsh.go (hunk 1 above)
Comment (zmb3) [reply tool/tsh/tsh.go:427]: Sure, happy to do that. These names are just from the canonical Go example, nothing I'm tied to.

**Review by rosstimothy — APPROVED** [review]
Good stuff!!

File: tool/tsh/tsh.go [hunk 2]
```diff
@@ -958,64 +974,80 @@ func serializeVersion(format string, proxyVersion string) (string, error) {
...
+//    recording by connecting to the Teleport cluster
 func onPlay(cf *CLIConf) error {
+	if format := strings.ToLower(cf.Format); format == teleport.PTY {
+		return playSession(cf)
+	}
+	return exportSession(cf)
+}
+
+func exportSession(cf *CLIConf) error {
 	format := strings.ToLower(cf.Format)
```
[the remark this answers is not in the document]
Comment (zmb3) [reply tool/tsh/tsh.go:998]: Not that I'm aware of, though to be honest I don't know why we do this with format and not any other flags.

**Pull request merged**